verify/trust/trust.go

```diff
@@ -223,16 +229,22 @@ func (r *ProductCerts) FromKDSCert(path string) error {
 	return r.FromKDSCertBytes(certBytes)
 }
 
-// X509Options returns the ASK and ARK as the only intermediate and root certificates of an x509
+// X509Options returns the AS[V]K and ARK as the only intermediate and root certificates of an x509
 // verification options object, or nil if either key's x509 certificate is not present in r.
-func (r *ProductCerts) X509Options(now time.Time) *x509.VerifyOptions {
+// The choice between ASK and ASVK is determined bey key.
+func (r *ProductCerts) X509Options(now time.Time, key abi.ReportSigner) *x509.VerifyOptions {
 	if r.Ask == nil || r.Ark == nil {
 		return nil
 	}
 	roots := x509.NewCertPool()
 	roots.AddCert(r.Ark)
 	intermediates := x509.NewCertPool()
-	intermediates.AddCert(r.Ask)
+	switch key {
+	case abi.VcekReportSigner:
+		intermediates.AddCert(r.Ask)
+	case abi.VlekReportSigner:
+		intermediates.AddCert(r.Asvk)
+	}
```

Comment:
Move the `r.Ask == nil` case from the `if` in L236 to the switch.
```suggestion
	switch key {
	case abi.VcekReportSigner:
	        if r.Ask == nil { 
	            return nil
	        }
		intermediates.AddCert(r.Ask)
	case abi.VlekReportSigner:
		if r.Asvk == nil { 
	            return nil
	        }
		intermediates.AddCert(r.Asvk)
	}
```
That was the only problem left during my testing :tada:

Reply:
Nice, thanks so much for your help testing. It's tough without a spec :P